parsec/core/gui/trio_thread.py

```diff
@@ -143,67 +120,23 @@ def set_exception(self, exc):
     def _set_done(self):
         self._done.set()
         signal = self._qt_on_success if self.is_ok() else self._qt_on_error
-        signal.emit(self) if signal.args_types else signal.emit()
+        if signal.signal.endswith("(PyQt_PyObject)"):
+            signal.emit(self)
+        else:
+            signal.emit()
 
-    def cancel_and_join(self):
-        assert self.cancel_scope
-        try:
-            trio.from_thread.run_sync(self.cancel_scope.cancel, trio_token=self._trio_token)
-        except trio.RunFinishedError:
-            pass
-        self._done.wait()
+    def cancel(self):
+        self.cancel_scope.cancel()
 
 
 class QtToTrioJobScheduler:
```

Comment:
In theory this whole job system is no longer needed right ? (it is just kept to avoid a huge refactoring of the codebase)

Reply:
We still need a nursery with some throttle logic (`submit_throttled_job`), so it's still useful. Wrapping trio tasks as job that we can retrieve or cancel also makes sense. The major cleanup would be getting rid of the `qt_on_success` and `qt_on_error` signals.